### PR TITLE
For #240: Implemented Python and Postgres availability check

### PR DESCRIPTION
**Pull request by ekondrashev**
For #240:  Implemented Python and Postgres availability check
* Added puzzle to compete the effort by adding redis availability check and define/cover rest of dependencies
* Added `venv` creation instruciton

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#247](https://github.com/timelesslounge/timelessis/issues/247) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#247](https://github.com/timelesslounge/timelessis/issues/247) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/9d169a51-30b2-418e-b433-f92873911865); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: deploy.sh [hunk 1]
```diff
@@ -45,6 +43,7 @@ ssh -i $STAGING_KEY $STAGING_USER@$STAGING_SERVER << EOF
   sudo -u postgres psql -c "CREATE DATABASE timelessdb_test;"
   echo "-- REPLACE: add scripts to cron"
   cd /app
+  python -m venv venv
```
Comment (vryazanov) [inline deploy.sh]: @ekondrashev see line 13, there we execute `pytest`, but only here after 30 lines of code we are creating env.
Looks strange to me. @emilianodellacasa is it ok that we run `pytest` so early?

**Comment by vryazanov** [comment]
@rultor good to merge

**Comment by vryazanov** [comment]
@emilianodellacasa I have no more comments, please have a look at the one I left

File: deploy.sh (hunk 1 above)
Comment (emilianodellacasa) [reply deploy.sh]: @vryazanov Actually it's extra, but main idea here is to check if all tests are green before deploying to staging server. In theory we do it when merging, but you'll never know

File: scripts/install/deploy/install_redis.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# @todo #240:30min Implement redis availability check and install flows. In case
+# Redis is not available lets install it and launch.
```
Comment (emilianodellacasa) [inline scripts/install/deploy/install_redis.sh]: @ekondrashev The puzzle is not well formatted: you should add a leading space for all lines AFTER the @todo

**Comment by emilianodellacasa** [comment]
@ekondrashev Just a comment on the puzzle format, fix it and I will merge

File: scripts/install/deploy/install_redis.sh (hunk 2 above)
Comment (ekondrashev) [reply scripts/install/deploy/install_redis.sh]: @emilianodellacasa fixed, thx

**Comment by ekondrashev** [comment]
@emilianodellacasa done with the puzzle and also cleaned redundant `echo`, please see last commits

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@ekondrashev puzzle format still not ok, please check

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by vryazanov** [comment]
@ypshenychka yes, I confirm that will try to find at least three major problem next time :)

**Comment by 0crat** [comment]
The job [#247](https://github.com/timelesslounge/timelessis/issues/247) is now out of scope

**Comment by ypshenychka** [comment]
@vryazanov Oh wow :) Proactive approach - I like that! But still need to hear your agreement on that quality will be bad, since no issues were found here. Agree?

**Comment by vryazanov** [comment]
> @vryazanov Oh wow :) Proactive approach - I like that! But still need to hear your agreement on that quality will be bad, since no issues were found here. Agree?

@ypshenychka yes

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vryazanov thanks

**Comment by ypshenychka** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**